workshop_code/workshop_3/Workshop3.ipynb

```diff
@@ -0,0 +1,565 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2018-10-19T21:50:41.887369Z",
+     "start_time": "2018-10-19T21:50:41.882884Z"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# Cross validation code that we won't run now, but you can try it in your own time\n",
+    "# This should take approximately 10 minutes to run (go grab dinner in between?)\n",
+    "# Recalling that C is inversely proportional to regularization strength\n",
+    "\n",
+    "'''\n",
+    "for i in range(0.3, 1.3, 0.1):\n",
+    "    LRmodel = LR(C=i, solver='lbfgs')\n",
+    "    LRmodel.fit(X_train, y_train)\n",
+    "    print('Logistic Regression Training Accuracy with C = {}: {}'\n",
+    "         .format(i, LRmodel.score(X_train, y_train)))\n",
+    "    print('Logistic Regression Validation Accuracy with C = {}: {}'\n",
+    "         .format(i, LRmodel.score(X_val, y_val)))\n",
+    "'''"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 3. Decision Trees on the Iris dataset"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2018-10-20T04:33:31.530210Z",
+     "start_time": "2018-10-20T04:33:31.502646Z"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "df_iris = ### FILL IN ###\n",
+    "df_iris.head(10)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2018-10-20T04:33:32.408967Z",
+     "start_time": "2018-10-20T04:33:32.403341Z"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "df_iris_X = ### FILL IN ###\n",
+    "df_iris_y = ### FILL IN ###\n",
+    "\n",
+    "X_train, X_val, X_test, y_train, y_val, y_test = split(df_iris_X, df_iris_y)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2018-10-20T04:33:33.199604Z",
+     "start_time": "2018-10-20T04:33:33.191350Z"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "clf = tree.DecisionTreeClassifier()\n",
+    "clf"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2018-10-20T04:33:34.707402Z",
+     "start_time": "2018-10-20T04:33:34.700669Z"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "clf = clf.fit(X_train, y_train)\n",
+    "print('Decision Tree train set accuracy: {}'.format(clf.score(X_train, y_train)))\n",
+    "print('Decision Tree validation set accuracy: {}'.format(clf.score(X_val, y_val)))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2018-10-20T04:35:50.424919Z",
+     "start_time": "2018-10-20T04:35:50.382090Z"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# dot_data = tree.export_graphviz(clf, out_file=None, \n",
```

Comment:
Add comment to what this does. 
why is it commented out?

Reply:
It doesn't work, I'm just leaving it there because the following image in the cell below it was generated using that code. The graphviz library is just difficult to set up